src/main/daml/Daml/Finance/Interface/Types/Common/Types.daml

```diff
@@ -93,3 +93,16 @@ data Quantity u a = Quantity
   deriving (Eq, Ord, Show)
 
 type InstrumentQuantity = Quantity InstrumentKey Decimal
+
+-- | Describes an underlying of a product.
+data Underlying = Underlying
```

Comment:
With weight and initialPrice this type feels a bit specific to sit in this rather generic package. Not sure there's a better place though.

Reply:
I first put it in the Swaps package, but that caused a dependency in the claims builder function/package on the Swaps package, which was not ideal (and caused a build problem due to the order that the packages are built). I could put it back in the Swaps package but extract the `Underlying` data and pass it on as a list of tuples (asset identifier and weight) to the claims builder function.
Or put this type somewhere else, of course.

Reply:
I think having it in the swaps package and changing the signature of the builder function is preferable here. If we end up having to change / evolve this type, I'd not want to be breaking the omni-present Common.Types package